src/visualization/src/Visualizer.cpp

```diff
@@ -552,4 +584,52 @@ bool Visualizer::isWindowActive() const
 #endif
 }
 
+
+bool Visualizer::setColorPalette(const std::string &name)
+{
+#ifdef IDYNTREE_USES_IRRLICHT
+
+    if( !pimpl->m_isInitialized )
+    {
+        reportError("Visualizer",
+                    "setColorPalette",
+                    "Impossible to set the color palette. Please initialize the visualizer");
+        return false;
+    }
+
+    const auto colors = pimpl->m_palette.find(name);
+
+    if(colors == pimpl->m_palette.end())
+    {
+        std::string paletteName;
+        for (const auto& tmp: pimpl->m_palette)
+            paletteName += " " + tmp.first;
+
+        const std::string error = "The palette named " + name
+                                + " does not exist. The following palette are available"
+                                + paletteName + ".";
+
+        reportError("Visualizer","setColorPalette", error.c_str());
+        return false;
+    }
+
+    this->enviroment().setBackgroundColor(irrlicht2idyntree(colors->second.background));
+    this->enviroment().setFloorGridColor(irrlicht2idyntree(colors->second.gridColor));
+
+    // delete the frame in the origin and create a new one
+    pimpl->m_environment.m_rootFrameNode->drop();
```

Comment:
```suggestion
    pimpl->m_environment.m_rootFrameNode->remove();
    pimpl->m_environment.m_rootFrameNode->drop();
```
The ``drop`` method corresponds to the destructor of a shared pointer. If the same objected is ``grab``bed somewhere else, it will not be removed. Instead, if the object calling ``drop`` is the last one having a pointer to that object, that object gets deallocated. In other words, ``drop`` just reduce by 1 the reference count. To make sure that the root frame node is removed from the scene, you have to call ``remove`` too.

Reply:
In ed7ea33 I commit your suggestion however If I run `idyntree-model-view` I got a segmentation fault at
```cpp
pimpl->m_environment.m_rootFrameNode->drop(); 
```

Reply:
My bad. I did not notice that ``remove`` calls ``drop``. Maybe, removing that line (the one with ``drop``) should be safe.